ospd_openvas/daemon.py

```diff
@@ -1401,20 +1401,66 @@ def exec_scan(self, scan_id: str):
                     )
 
                     if scan_db.host_is_finished(openvas_scan_id):
-                        self.report_openvas_scan_status(
-                            scan_db, scan_id, current_host
+                        # Check that the host status is updated. As it has
+                        # finished, its stauts should be DEAD or FINISHED.
+                        # Otherwise it is considered an error. It tries 3
+                        # times until considered it an error
+                        retry = 3
+                        host_progress = self.get_scan_host_progress(
+                            scan_id, current_host
                         )
+                        while (
+                            host_progress > -1 and host_progress < 100
+                        ) and retry > 0:
+                            logger.debug(
+                                '%s: Host %s finished but its scan progress is '
+                                'still %d. Retrying the host progress update '
+                                'in 1 second',
+                                scan_id,
+                                current_host,
+                                host_progress,
+                            )
+                            time.sleep(1)
+                            self.report_openvas_scan_status(
+                                scan_db, scan_id, current_host
+                            )
+                            retry -= 1
+                            host_progress = self.get_scan_host_progress(
+                                scan_id, current_host
+                            )
 
+                        if retry == 0:
```

Comment:
Even if after the third retry we successfully get a valid host_progress the `retry` will be set to 0 nonetheless and an error is logged. Maybe a `break` can be used in the while loop.